File: algorithms/1_easy/E/01287_element_appearing_more_than_25_in_sorted_array/solution.c
int findSpecialInteger(const int *restrict arr, const int len) {
    const int target_cnt = len / 4;
    int cnt = 0;
    int curr = 0;

    for (int i = 0; i < len; i += 1) {
        if (curr != arr[i]) {
            curr = arr[i];
            cnt = 1;
        } else {
            cnt += 1;
        }

        if (cnt > target_cnt) {
            return curr;
        }
    }

    perror("invvalid input");
    exit(1);
}

File: algorithms/1_easy/E/01287_element_appearing_more_than_25_in_sorted_array/solution.rs
impl Solution {
    pub fn find_special_integer(arr: Vec<i32>) -> i32 {
        let target_cnt: usize = arr.len() / 4;
        let mut cnt: usize = 0;
        let mut curr: i32 = 0;

        for a in arr {
            if curr != a {
                curr = a;
                cnt = 1;
            } else {
                cnt += 1;
            }

            if cnt > target_cnt {
                return curr;
            }
        }

        unreachable!()
    }
}
